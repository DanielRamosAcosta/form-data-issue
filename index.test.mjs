import { test, expect } from "vitest";
import { FormData } from "formdata-node";
import {
  toStringRepresentation,
  removeRandomnessOfForm,
  crlfToLf,
  createReadableStream,
} from "./utils.mjs";

test("add files", async () => {
  const form = new FormData();
  const stream = createReadableStream("Stream content");

  form.set("stream", {
    type: "text/plain",
    name: "file.txt",
    [Symbol.toStringTag]: "File",
    stream() {
      return stream;
    },
  });

  const formString = await toStringRepresentation(form)
    .then(removeRandomnessOfForm)
    .then(crlfToLf);
  expect(formString).toMatchSnapshot();
});
