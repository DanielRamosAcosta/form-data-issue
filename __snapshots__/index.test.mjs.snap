// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`add files 1`] = `
"--form-data-boundary-xxxxxxxxxxxxxxxx
Content-Disposition: form-data; name=\\"stream\\"; filename=\\"file.txt\\"
Content-Type: text/plain

Stream content
--form-data-boundary-xxxxxxxxxxxxxxxx--

"
`;
